src/pure/PureMicroPasses.ml

```diff
@@ -3635,6 +3635,20 @@ let add_type_annotations_to_fun_decl (trans_ctx : trans_ctx)
                 log#ldebug
                   (lazy (__FUNCTION__ ^ ": function name: " ^ trans_fun.name));
                 trans_fun.signature
+            | TraitMethod (tref, method_name, method_decl_id) ->
+                log#ldebug
+                  (lazy (__FUNCTION__ ^ ": method name: " ^ method_name));
+                if Option.is_some lp_id then
+                  craise __FILE__ __LINE__ span
+                    "Trying to get a loop subfunction from a method call";
+                let method_sig =
+                  silent_unwrap __FILE__ __LINE__ span
+                    (Charon.Substitute.lookup_flat_method_sig trans_ctx.crate
+                       tref.trait_decl_ref.trait_decl_id method_name)
+                in
```

Comment:
Could you add a TODO here? This works, but there should be a way of not invoking the translation there (by looking up the *pure* trait declaration)

Reply:
> We'd have to precompute these method signatures and store them somewhere

Yes, that's exactly what I have in mind: we would update the pure trait_decl definitino.